src/components/SearchFacet/FacetList.tsx

```diff
@@ -128,14 +164,96 @@ export const NodeList = memo((props: INodeListProps) => {
     }
   };
 
+  const handleKeyboardFocusNext = (index: number[]) => {
```

Comment:
All this logic should probably be moved into the facetStore, and just have action calls on keyboard events.